tangos/input_handlers/yt.py

```diff
@@ -85,6 +170,130 @@ def _load_halo_cat_without_caching(self, ts_extension, snapshot_file):
     def get_properties(self):
         return {}
 
+    def available_object_property_names_for_timestep(self, ts_extension, object_typetag):
+        h, _ = self._load_halo_cat(ts_extension, object_typetag)
+        return [fn for ft, fn in h.field_list if ft == "halos"]
+
+
+    def iterate_object_properties_for_timestep(self, ts_extension, object_typetag, property_names):
+        try:
+            yield from super().iterate_object_properties_for_timestep(ts_extension, object_typetag, property_names)
+            return
+        except OSError:
+            pass
+        h, ad = self._load_halo_cat(ts_extension, object_typetag)
+
+        props_with_ftype = [
+            ("halos", name) for name in property_names
+        ]
+
+        ad.get_data(props_with_ftype)
+
+        Nhalo = len(ad["halos", "particle_identifier"])
+        yield from zip(range(Nhalo), range(Nhalo), *(
+            ad[_] for _ in props_with_ftype
+        ))
+
+
+class YtRamsesRockstarInputHandler(YtInputHandler):
+    patterns = ["output_0????"]
+    auxiliary_file_patterns = ["halos_*.bin"]
+
+    def load_timestep_without_caching(self, ts_extension, mode=None):
+        from yt.data_objects.particle_filters import add_particle_filter
+        if mode is not None:
+            raise ValueError("Custom load modes are not supported with yt")
+        f = yt.load(self._extension_to_filename(ts_extension))
+
+        def Stars(pfilter, data):
+            filter = data[("all", "particle_family")] == 2 # DM = 1, Stars = 2
+            return filter
+
+        add_particle_filter("stars", function=Stars, filtered_type='all', \
+                            requires=["particle_family"])
+
+        def AllDarkMatter(pfilter, data):
+            #filter = np.logical_or(data[("all", "particle_type")] == 4,data[("all", "particle_type")] == 1) # DM = 1, Stars = 2
+            filter = data[("all", "particle_family")] == 1
+            return filter
+
+        add_particle_filter("dark_matter", function=AllDarkMatter, filtered_type='all', \
+                            requires=["particle_family"])
+
+        def MustRefineParticles(pfilter, data):
+            filter = data[("all", "particle_family")] == 4
+            return filter
+
+        add_particle_filter("mrp_dark_matter", function=MustRefineParticles, filtered_type='all', \
+                            requires=["particle_family"])
+
+        f.add_particle_filter("stars")
+        f.add_particle_filter("dark_matter")
+        f.add_particle_filter("mrp_dark_matter")
+
+        return f
```

Comment:
For the past 6 years, yt automatically creates those filters if the `particle_family` array already exists, so those lines are actually mostly redundant (and should be yt-specific anyways). I'd suggest
```suggestion
        if mode is not None:
            raise ValueError("Custom load modes are not supported with yt")
        return yt.load(self._extension_to_filename(ts_extension))
```